### PR TITLE
Allows unanchoring and deconstructing of linen bins

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request

Makes linen bins removable without resorting to explosives.

## Why It's Good For The Game

image

## Changelog
:cl:
tweak: Linen bins can now be (un)achored with a wrench, or disassembled with a screwdriver (when empty).
add: You can now make linen bins using two rods.
/:cl:

I dunno, it was weird that there was no path for moving or deconstructing these. You can't even attack it since attacks are intercepted by attempts to pick up bedsheets (or hide things, apparently).

### CODE REVIEW

File: code/game/objects/structures/bedsheet_bin.dm [hunk 1]
```diff
@@ -322,6 +322,17 @@ LINEN BINS
 		amount++
 		to_chat(user, "<span class='notice'>You put [I] in [src].</span>")
 		update_icon()
+	else if(istype(I, /obj/item/wrench) && I.use_tool(src, user, 5, volume=50))
```
Comment (AnturK) [inline code/game/objects/structures/bedsheet_bin.dm]: Use the proper helpers here, default_unfasten_wrench etc.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/bedsheet_bin.dm (hunk 1 above)
Comment (Barhandar) [reply code/game/objects/structures/bedsheet_bin.dm]: also I.tool_behaviour == TOOL_WRENCH etc reeee.

File: code/game/objects/structures/bedsheet_bin.dm [hunk 2]
```diff
@@ -322,6 +322,17 @@ LINEN BINS
 		amount++
 		to_chat(user, "<span class='notice'>You put [I] in [src].</span>")
 		update_icon()
+	else if(istype(I, /obj/item/wrench) && I.use_tool(src, user, 5, volume=50))
+		anchored = !anchored
+		to_chat(user, "<span clas='notice'>You [anchored ? "" : "un"]wrench the [src].</span>")
+	else if(istype(I, /obj/item/screwdriver))
```
Comment (Barhandar) [inline code/game/objects/structures/bedsheet_bin.dm]: same, use tool_behaviour not typechecks

**Comment by PKPenguin321** [comment]
Check for non-help intent to deconstruct so you can still hide these items in these bins if you're on help intent

Also maybe allow these to be constructed out of rods

**Comment by zxaber** [comment]
How often do you need to hide a wrench or screwdriver in a linen bin?

**Comment by PKPenguin321** [comment]
Shouldn't matter

File: code/game/objects/structures/bedsheet_bin.dm (hunk 1 above)
Comment (zxaber) [reply code/game/objects/structures/bedsheet_bin.dm]: Done for wrench. Screwdriver doesn't have a default proc that I can see (only one for opening panels), so I just changed it to tool behavior.

File: code/game/objects/structures/bedsheet_bin.dm (hunk 2 above)
Comment (zxaber) [reply code/game/objects/structures/bedsheet_bin.dm]: Done

**Review by AnturK — CHANGES_REQUESTED** [review]
Just some minor stuff left

File: code/game/objects/structures/bedsheet_bin.dm [hunk 3]
```diff
@@ -314,6 +319,8 @@ LINEN BINS
 		update_icon()
 	..()
 
+//obj/structure/bedsheetbin/default_deconstruction_screwdriver(mob/user, obj/item/I, time = 10)
```
[the remark this answers is not in the document]
Comment (zxaber) [reply code/game/objects/structures/bedsheet_bin.dm]: Oof, my bad. Fixed

File: code/game/objects/structures/bedsheet_bin.dm [hunk 4]
```diff
@@ -322,6 +329,19 @@ LINEN BINS
...
+	else if(default_unfasten_wrench(user, I, 5))
+		return
+
+	else if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		if(amount)
+			to_chat(user, "<span clas='warn'>The [src] must be empty first!</span>")
+			return
+		if(I.use_tool(src, user, 5, volume=50))
+			to_chat(user, "<span clas='notice'>You disassemble the [src].</span>")
+			new /obj/item/stack/rods(loc, 2)
```
[the remark this answers is not in the document]
Comment (zxaber) [reply code/game/objects/structures/bedsheet_bin.dm]: Done and done.

**Pull request merged**